### PR TITLE
fix(pagetitlebar): fix some alignment/spacing issues

### LINKED ISSUE #2143: [Page header] Design specs
### What package is this for?

- [x] React
- [ ] Angular

[Page header - spec sheet.pdf](https://github.com/carbon-design-system/carbon-addons-iot-react/files/6227336/Page.header.-.spec.sheet.pdf)
Page header

### Available extra resources

Component guidance reference:
https://pages.github.ibm.com/ai-applications/design/components/page-header/usage

### LINKED ISSUE #2144: [Page header] Design specs
### What package is this for?

- [ ] React
- [x] Angular

[Page header - spec sheet.pdf](https://github.com/carbon-design-system/carbon-addons-iot-react/files/6227340/Page.header.-.spec.sheet.pdf)

Page header

### Available extra resources

Component guidance reference:
https://pages.github.ibm.com/ai-applications/design/components/page-header/usage

### PR DESCRIPTION
**Pull request by jkeen**
Closes #2143  

**Summary**

- Verified colors and spacing in design doc and corrected irregularities according to [the spec sheet](https://github.com/carbon-design-system/carbon-addons-iot-react/files/6227340/Page.header.-.spec.sheet.pdf)
- Added missing states/behaviors listed in the design doc

**Change List (commits, features, bugs, etc)**

- use css grid to align page header, ensuring actions and buttons are always in line regardless of surrounding options
- trigger and progress transitions gradually during scroll
- simplify PageTitleBar template, relying mainly on css modifier classes on the root component node
- added area for upper actions and transition as described [here](https://pages.github.ibm.com/ai-applications/design/components/page-header/usage/)
- truncate page header title and breadcrumb title (in condensed mode) correctly
- adjusted alignment of tabs in relation to title
- made accommodation for when `<InlineLoading>` is used as description
- fixed RTL layout issues
- made sure spacing between buttons and search, selects, or other non-button objects in the toolbar have 16px margin.
- single primary buttons should have a left margin, but primary + secondary next to each other should not have space between them
- added new stories illustrating key scenarios, with other knobs for additional testing

Issues to resolve: The PageTItleBar padding should be what the specs indicate (correct), but the correction change threw off a little alignment in the PageWizard story.

**Acceptance Test (how to verify the PR)**

- Compare relevant sections of the spec sheet with the output described above

### CODE REVIEW

**Review by JoelArmendariz — DISMISSED** [review]
LGTM

**Comment by davidicus** [comment]
Can we adjust the delay on the background color opacity? It takes too long and you can see the overlay before the background fills in.
https://user-images.githubusercontent.com/1590966/114428877-bd616180-9b8a-11eb-9dc5-c4fb2a0024f4.mov

**Comment by davidicus** [comment]
In the issue for this PR the designer state the tabs should have the sticky effect instead of the header. It looks like it is always the header even with Tabs present.

**Comment by davidicus** [comment]
The padding on top of breadcrumbs seems larger than the 12px of padding called for in the spec.

**Comment by davidicus** [comment]
There seems to be several places where the spec is not being respected in this story [here](https://deploy-preview-2208--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-pagetitlebar--with-dynamic-scrolling). Tooltip icon does not have the proper size and there is some spacing issues.

**Review by davidicus — CHANGES_REQUESTED** [review]
There are a few items that still need to be addressed.

**Comment by davidicus** [comment]
@Najil-K @dianatran18 Please review the PR for alignment.

**Comment by dianatran18** [comment]
Hey @davidicus should which one should we be reviewing specifically?

**Comment by davidicus** [comment]
@dianatran18 [This](https://deploy-preview-2208--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-pagetitlebar--with-everything) story and [this](https://deploy-preview-2208--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-pagetitlebar--with-dynamic-scrolling) story would be good to check. The second one you have to adjust the knobs for the different forms of sticky header.

**Comment by Najil-K** [comment]
**UI audit sheet:**

[UI audit - Page header.pdf](https://github.com/carbon-design-system/carbon-addons-iot-react/files/6316533/UI.audit.-.Page.header.pdf)

UI audit - Page Title (v2)@2x



@davidicus @dianatran18

**Review by davidicus — CHANGES_REQUESTED** [review]
This looks good but we still need to fix the tooltip icon outlined in issue 5 outlined by @Najil-K.  FYI @Najil-K issue 4 is showing the focus blue ring which Carbon actually has in their storybook as well.

File: packages/react/src/components/PageTitleBar/PageTitleBar.jsx [hunk 1]
```diff
@@ -76,8 +78,7 @@ const defaultProps = {
   tabs: undefined,
   content: undefined,
   headerMode: HEADER_MODES.STATIC,
-  stickyHeaderOffset: 48, // default to 3rem to stick to the bottom of the suite header
-  dynamicTransitionOffset: 48, // default to 3rem to stick to the bottom of the suite header
+  stickyHeaderOffset: 0, // default to 3rem to stick to the bottom of the suite header
```
Comment (davidicus) [inline packages/react/src/components/PageTitleBar/PageTitleBar.jsx:81]: Does this comment still apply here?

**Review by davidicus — APPROVED** [review]

**Pull request merged**